pkg/route/validation/validation.go

```diff
@@ -90,6 +101,33 @@ func validateRoute(route *routev1.Route, checkHostname bool) field.ErrorList {
 		result = append(result, err)
 	}
 
+	if len(route.Spec.HTTPHeaders.Actions.Response) != 0 || len(route.Spec.HTTPHeaders.Actions.Request) != 0 {
+		if route.Spec.TLS != nil && route.Spec.TLS.Termination == routev1.TLSTerminationPassthrough {
+			specTLS := field.NewPath(specTLSTerminationString)
+			result = append(result, field.NotSupported(specTLS, route.Spec.TLS.Termination, []string{"Only edge and re-encrypt routes are supported for providing customized headers."}))
+		}
+	}
+	if len(route.Spec.HTTPHeaders.Actions.Response) != 0 && len(route.Spec.HTTPHeaders.Actions.Response) > 20 {
```

Comment:
```suggestion
	if len(route.Spec.HTTPHeaders.Actions.Response) > 20 {
```
Do we need both comparisons? `len(...) > 20` implies `len(...) != 0`

Reply:
no